libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -438,111 +470,30 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     }
   }
 
-  private readonly elementToParentMap: { [key: string]: () => HTMLElement[] } = {
-    'KIRBY-MODAL-FOOTER': () => [this.elementRef.nativeElement],
-    'KIRBY-PAGE-TITLE': () =>
-      [this.ionTitleElement.nativeElement, this.contentTitle?.nativeElement].filter(
-        (element) => element !== undefined
-      ),
-    'KIRBY-PAGE-PROGRESS': () => [this.ionToolbarElement.nativeElement],
-  };
-
-  private clearEmbeddedElements() {
-    Object.entries(this.elementToParentMap).forEach(([tagName, getParents]) => {
-      const newParents = getParents();
-      newParents.forEach((newParent) => {
-        const embeddedElement = newParent.querySelector<HTMLElement>(`:scope > ${tagName}`);
-        this.removeChild(embeddedElement);
-      });
-    });
-  }
-
-  /* TODO: Rewrite to make this function independent of element order. 
-     See: https://github.com/kirbydesign/designsystem/issues/2096
-  */
-  private getEmbeddedComponentElement(): null | Element {
-    const contentElementChildren = Array.from(
-      this.ionContentElement.nativeElement.children
-    ).reverse(); // Reverse makes it easier to retrieve the last children in the list
-
-    const embeddedComponentElement = !!this.config.modalRoute
-      ? contentElementChildren[0]
-      : contentElementChildren[1];
-
-    /* 
-      As ModalConfig.component has type 'any' all values are valid for component; 
-      explicitly handle the case where no embedded component element is found due to 
-      this.
-    */
-    if (!embeddedComponentElement) return null;
-    return embeddedComponentElement;
-  }
-
   private getEmbeddedFooterElement() {
-    return this.elementRef.nativeElement.querySelector<HTMLElement>('kirby-modal-footer');
+    return this.currentFooter;
   }
 
-  private moveChild(child: Element, newParents: Element[]) {
+  private moveChild(
+    childElementRef: ElementRef<HTMLElement>,
+    newParentElementRef: ElementRef<HTMLElement>
+  ) {
+    const child = childElementRef.nativeElement;
+    const newParent = newParentElementRef.nativeElement;
     this.renderer.removeChild(child.parentElement, child);
-
-    newParents.forEach((newParent, index) => {
-      const childToAppend = index > 0 ? child.cloneNode(true) : child;
-      this.renderer.appendChild(newParent, childToAppend);
-      // Append adds child as last element of parent; therefore retrieve with lastElementChild
-      const childElement = newParent.lastElementChild;
-
-      if (childElement.tagName === 'KIRBY-MODAL-FOOTER') {
-        this.resizeObserverService.observe(childElement, (entry) => {
-          const [property, pixelValue] = [
-            '--footer-height',
-            `${Math.floor(entry.contentRect.height)}px`,
-          ];
-          this.setCssVar(this.elementRef.nativeElement, property, pixelValue);
-        });
-      }
-    });
+    this.renderer.appendChild(newParent, child);
```

Comment:
To simplify the method it has been changed to only accomodate a single new parent (check type signature on line 479)